coral-common/src/main/java/com/linkedin/coral/common/transformers/StandardUDFOperatorTransformer.java

```diff
@@ -0,0 +1,260 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.common.transformers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.OperandTypes;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SqlOperandTypeChecker;
+import org.apache.calcite.sql.type.SqlReturnTypeInference;
+import org.apache.calcite.sql.validate.SqlUserDefinedFunction;
+
+import com.linkedin.coral.com.google.common.base.Preconditions;
+import com.linkedin.coral.common.functions.FunctionReturnTypes;
+
+import static com.linkedin.coral.common.calcite.CalciteUtil.*;
+
+
+/**
+ * This class is the base class which transforms UDF operator with the rules defined as JSON string on SqlNode layer
+ */
+public class StandardUDFOperatorTransformer extends SqlCallTransformer {
```

Comment:
Let us use the UDF term for actual user defined functions. It was wrongly used in the previous Coral versions, but we should start fixing this. Also `Standard` is a bit generic, so let us come up with a more descriptive name of this class. Since this bases its condition on name + number of arguments, maybe something that says `Signature` works? There might be other names. We can recommend options offline.

Reply:
Good point, I will try to figure out a name more specific and not related UDF.

Reply:
I change the name to `SignatureBasedConditionSqlCallTransformer`